functional-test/src/test/groovy/org/rundeck/tests/functional/selenium/project/ExecutionModeSpec.groovy

```diff
@@ -207,9 +207,9 @@ class ExecutionModeSpec extends SeleniumBase{
         waitFor(ExecutionUtils.Retrievers.executionsForProject(client, projectName),
                 { List<Execution> execs ->  execs.any(ExecutionUtils.Verifiers.executionRunning()) },
                 WaitingTime.EXCESSIVE)
-        // Waits for at least one execution to start running
+        // Waits for all executions to finish
         waitFor(ExecutionUtils.Retrievers.executionsForProject(client, projectName),
-                { List<Execution> execs ->  execs.any(ExecutionUtils.Verifiers.executionFinished()) },
+                verifyForAll(ExecutionUtils.Verifiers.executionFinished()),
```

Comment:
I can see how this fix would make this test _less flaky_ but i'm still a little concerned that this job will be producing new executions the longer this portion of the test runs.

Ideally it seems like this test needs a way to do something closer to: "wait for at least one new execution to start and complete"?

cc: @ronaveva.

Reply:
I think it's fine if it produces new executions too, I believe the flakiness issue was due to it passing this waitFor check *before* the later expected condition was true (number of completed executions is greater than it was before on line 216). As long as it waits for 1 execution to complete, that should satisfy the later check.

Reply:
if someone wants to refactor this test case (and the one after it) to be more exact, that's worth a pull request

Reply:
> I believe the flakiness issue was due to it passing this waitFor check before the later expected condition was true (number of completed executions is greater than it was before on line 216). 

Yes I see what you are saying and that makes sense.

> As long as it waits for 1 execution to complete, that should satisfy the later check.

Yes that is what I am trying to say in my comment above as well. My concern is that the job is starting a new execution every `*/2` seconds and this assertion code is waiting for _all_ executions to be finished. This feels like another potential race condition where we could tighten up the code.

Let's merge it and see how it goes though.